### PR TITLE
Level Builder: Fixed width & height and refactored

### PR DESCRIPTION
**Pull request by OlafSzmidt**
Fixed issue #40 with this and re factored some of the code.

### CODE REVIEW

**Comment by ramonfmir** [comment]
I like the GUIUtils idea. Maybe even more code from other menus could be moved there.

**Comment by OlafSzmidt** [comment]
I suppose that can be done alongside #48.

File: UnityProject/Assets/Editor/LevelBuilder/Util/GUIUtils.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
```
Comment (CelineBoudier) [inline UnityProject/Assets/Editor/LevelBuilder/Util/GUIUtils.cs]: can we check if all these are necessary? :)

Comment (ramonfmir) [reply UnityProject/Assets/Editor/LevelBuilder/Util/GUIUtils.cs]: I am quite sure that in most files we have unnecessary imports... I opened #65

File: UnityProject/Assets/Editor/LevelBuilder/Util/GUIUtils.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	// with no space between the label and the textfield.
+	public static string TextField(string label, string text)
+	{
+		var textDimensions = GUI.skin.label.CalcSize (new GUIContent (label));
+		EditorGUIUtility.labelWidth = textDimensions.x;
+
+		return EditorGUILayout.TextField (label, text);
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (OlafSzmidt) [reply UnityProject/Assets/Editor/LevelBuilder/Util/GUIUtils.cs]: Yup, it is. The whole codebase is converted to UNIX.

File: UnityProject/Assets/Editor/LevelBuilder/Util/GUIUtils.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	// A utility method that takes creates EditorGUILayout.TextField
+	// with no space between the label and the textfield.
+	public static string TextField(string label, string text)
+	{
+		var textDimensions = GUI.skin.label.CalcSize (new GUIContent (label));
+		EditorGUIUtility.labelWidth = textDimensions.x;
+
+		return EditorGUILayout.TextField (label, text);
+	}
+}
```
Comment (CelineBoudier) [inline UnityProject/Assets/Editor/LevelBuilder/Util/GUIUtils.cs]: new line please? @ramonfmir might notice :D

Comment (OlafSzmidt) [reply UnityProject/Assets/Editor/LevelBuilder/Util/GUIUtils.cs]: Realised it was deleted after my commit - just fixed and rebased them! :D

File: UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GridMenu.cs [hunk 4]
```diff
@@ -8,43 +8,39 @@
...
-			catch (FormatException exception) 
-			{
-				Debug.Log (
-					"Grid width/height format exception." +
-					"Please make sure you write an integer.");
-			}
-		else
-			GridVisualiser.DestroyGrid();
+		if (showGrid) try {
+			GridVisualiser.BuildGrid(Convert.ToInt32(width), Convert.ToInt32(height));
```
Comment (CelineBoudier) [inline UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GridMenu.cs]: hurgh- will go with #66 though ^^

File: UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GridMenu.cs [hunk 5]
```diff
@@ -8,43 +8,39 @@
...
+		}
+		catch(FormatException exception) {
+			Debug.Log("Grid width/height format exception." + "Please make sure you write an integer.");
```
Comment (CelineBoudier) [inline UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GridMenu.cs]: Not sure teenagers will know what an integer is!

Comment (OlafSzmidt) [reply UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GridMenu.cs]: The unity level builder will only be used by level designers, changed anyways!

**Comment by CelineBoudier** [comment]
Please squash and I'll approve :)

File: UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GridMenu.cs (hunk 5 above)
Comment (CelineBoudier) [reply UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GridMenu.cs]: That's fine then, but we need to be more descriptive when talking about "users" I think

Comment (OlafSzmidt) [reply UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GridMenu.cs]: Definitely:) I'll keep that in mind

**Comment by OlafSzmidt** [comment]
Done:)

**Pull request merged**